src/core/textures/BaseTexture.js

```diff
@@ -754,4 +768,72 @@ export default class BaseTexture extends EventEmitter
         // lets assume its a base texture!
         return source;
     }
+
+    /**
+     * Adds a BaseTexture to the global BaseTextureCache. This cache is shared across the whole PIXI object.
+     *
+     * @static
+     * @param {PIXI.BaseTexture} baseTexture - The BaseTexture to add to the cache.
+     * @param {string} id - The id that the BaseTexture will be stored against.
+     */
+    static addToCache(baseTexture, id)
+    {
+        if (id)
+        {
+            if (!baseTexture.textureCacheId)
+            {
+                baseTexture.textureCacheId = id;
+            }
+
+            // @if DEBUG
+            /* eslint-disable no-console */
+            if (BaseTextureCache[id])
+            {
+                console.warn(`BaseTexture added to the cache with an id [${id}] that already had an entry`);
+            }
+            /* eslint-enable no-console */
+            // @endif
+
+            BaseTextureCache[id] = baseTexture;
+        }
+    }
+
+    /**
+     * Remove a BaseTexture from the global BaseTextureCache.
+     *
+     * @static
+     * @param {string|PIXI.BaseTexture} baseTexture - id of a BaseTexture to be removed, or a BaseTexture instance itself.
+     * @return {PIXI.BaseTexture|null} The BaseTexture that was removed.
+     */
+    static removeFromCache(baseTexture)
+    {
+        if (typeof baseTexture === 'string')
+        {
+            const baseTextureFromCache = BaseTextureCache[baseTexture];
+
+            if (baseTextureFromCache)
+            {
+                baseTextureFromCache.textureCacheId = null;
+                delete BaseTextureCache[baseTexture];
+
+                return baseTextureFromCache;
+            }
+        }
+        else
+        {
+            for (const prop in BaseTextureCache)
```

Comment:
Same questions as the Texture cache, can we store all the ids on the object and just do lookups here rather than a loop.